### PR TITLE
When github application is suspended don't treat it as transient error

### PR DESCRIPTION
**Pull request by rcerven**
[STONEBLD-2218](https://issues.redhat.com//browse/STONEBLD-2218)

### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

### CODE REVIEW

File: controllers/component_build_controller_pac.go [hunk 1]
```diff
@@ -871,6 +871,11 @@ func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context
 		IsAppInstallationExpected: true,
 	})
 	if err != nil {
+		if strings.Contains(err.Error(), "installation has been suspended") {
+			log.Info(fmt.Sprintf("Installation has been suspended for git repository: %s", repoUrl))
```
Comment (mmorhun) [inline controllers/component_build_controller_pac.go:875]: Logs reader is not in context of the code, so please describe installation of what and where has been suspended. And probably consequences for the Component.

File: controllers/component_build_controller_pac.go [hunk 2]
```diff
@@ -871,6 +871,11 @@ func (r *ComponentBuildReconciler) ConfigureRepositoryForPaC(ctx context.Context
 		IsAppInstallationExpected: true,
 	})
 	if err != nil {
+		if strings.Contains(err.Error(), "installation has been suspended") {
```
Comment (mmorhun) [inline controllers/component_build_controller_pac.go:874]: Does it make sense to check the condition inside the git package like for other errors, e.g. [this](https://github.com/redhat-appstudio/build-service/blob/main/pkg/git/github/github_app.go#L61)?

Comment (rcerven) [reply controllers/component_build_controller_pac.go:874]: 403 is general forbidden, so I'd rather check error message

Comment (mmorhun) [reply controllers/component_build_controller_pac.go:874]: Yes, we definitely need to check error message, my point was in the place of the check.

Comment (rcerven) [reply controllers/component_build_controller_pac.go:874]: ok I'll check in github package

**Review by mmorhun — APPROVED** [review]

**Pull request merged**